### PR TITLE
12262 object detection tutorial

### PR DESCRIPTION
**Pull request by nengelmann**
Adding an object detection tutorial.
Upgrading to `Sphinx==5.0.0'`

This is how the documentation will look like:
[Object Detection — Konfuzio documentation.pdf](https://github.com/konfuzio-ai/konfuzio-sdk/files/15434561/Object.Detection.Konfuzio.documentation.pdf)

### CODE REVIEW

File: docs/sdk/tutorials.rst [hunk 1]
```diff
@@ -113,18 +113,23 @@ Learn how to dynamically generate PDF forms using our AI-powered tools. This tut
 
 Harness the power of regular expressions for precise Document Annotations. This tutorial guides you through the process of using regex patterns effectively.
 
+19. :doc:`Object Detection <tutorials/object-detection/index>`
```
[the remark this answers is not in the document]
Comment (nengelmann) [reply docs/sdk/tutorials.rst:116]: Done, will push soon. 
What effect does this have?

File: docs/sdk/tutorials/object-detection/index_nb.md [hunk 2]
```diff
@@ -0,0 +1,1026 @@
...
+
+Here is an example of what the model will be capable of after training. The color cyan stands for detected checked boxes and the color green for detected empty boxes, while the numbers indicate the detection confidence. 😃
+
+![Example of checkbox detection - 1](checkbox_example_handwritten_1.png)
+
+### Dependency installation 💿
+
+For training and testing the model, the following dependencies are needed.
+
+```bash tags=["remove-output"]
```
[the remark this answers is not in the document]
Comment (nengelmann) [reply docs/sdk/tutorials/object-detection/index_nb.md:61]: I feel like the multiple lines gives a quicker glance on what libraries are installed and used, so therefore I would like to keep as is. :)

File: docs/sdk/tutorials/object-detection/index_nb.md [hunk 3]
```diff
@@ -0,0 +1,1026 @@
...
+
+![Example of use case - 1](checkbox_example_use_case.png)
+
+
+### Conclusion 💭
+In this tutorial, we have trained, optimized and tested the object detection model YOLO-NAS on a coco dataset. Below is the full code to accomplish this task:
+
+
+**Dependency installation**
+```bash tags=["skip-execution", "nbval-skip"]
```
[the remark this answers is not in the document]
Comment (nengelmann) [reply docs/sdk/tutorials/object-detection/index_nb.md:615]: This is the conclusion part. Here I just copied the above code as a reference for the user to simply copy paste it. Therefore the code should not be executed in our pipeline again. 
The user of cause needs to install those libraries.

File: docs/sdk/tutorials/object-detection/index_nb.md [hunk 4]
```diff
@@ -0,0 +1,1026 @@
...
+## Object Detection
+
+---
+
+**Prerequisites:**
+
+- General understanding of object detection and training of neural networks.
+- A COCO formatted object detection dataset.
+
+**Difficulty:** Medium
```
[the remark this answers is not in the document]
Comment (nengelmann) [reply docs/sdk/tutorials/object-detection/index_nb.md:24]: Okay, I share your opinion. 

As I understood there is `Easy`, M̀edium`and `Hard` as difficulty level. So I add `Hard` in this case. 
There was also one single "`Intermediate`" in the Tutorials, I changed this to Medium as well.

File: docs/sdk/tutorials.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (iftwigs) [reply docs/sdk/tutorials.rst:116]: sphinx renders URIs and nested structure of the toctree correctly. if you look at the URLs of pages in dev documentation, you will see that there is the nested structure that corresponds to the paths mentioned in the toctree.

File: docs/sdk/tutorials/object-detection/index_nb.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (iftwigs) [reply docs/sdk/tutorials/object-detection/index_nb.md:636]: i should've added this comment in the first cell where you explain that this needs to be installed. my question is, how this is to be run in our documentation-building pipeline if you do not install these libraries in the environment beforehand? it is going to fail.

[the remark this answers is not in the document]
Comment (nengelmann) [reply docs/sdk/tutorials/object-detection/index_nb.md:636]: Just checked this locally again. So the first cell for the installation of the dependencies is just decorated with the tag remove output. So the cell gets executed while the notebook is ran by the test pipeline and the dependencies are installed. There is no library installed beforehand. 

https://github.com/konfuzio-ai/konfuzio-sdk/blob/3d12a1f96bfab1d2ab975513ca3fa6abb69144d7/docs/sdk/tutorials/object-detection/index_nb.md?plain=1#L61

**Review by iftwigs — APPROVED** [review]

**Review by zypriafl — APPROVED** [review]

**Pull request merged**